### DISCUSSION TITLE
Taking into account both T gates and Toffoli gates in the cost estimation

### DISCUSSION THREAD

**Discussion by PabloAMC** [quantumlib/OpenFermion#1376] [Features and ideas]
Right now only Toffoli gates are taken into account for the estimation of time and physical qubits required to synthesize non-Clifford gates in the circuit. 
https://github.com/quantumlib/OpenFermion/blob/cf53c063d0f124a02ff8776bb7f8afb110d4bde6/src/openfermion/resource_estimates/surface_code_compilation/physical_costing.py#L176-L184
From a Toffoli gate, 2 T gates might be obtained (https://quantum-journal.org/papers/q-2019-04-30-135/pdf/). However, the cost is not exactly 1->2 as the conversion takes time etc. I think adding T gates would be a great idea, in part because T gates are more natural when it comes to rotations, even if it seems that all state-of-the-art methods heavily rely on Toffolis.

**Comment by Ailol** [comment]
well toffo is actualy nils torfinn balto(ove kåvens group...) i shed a tear for all the babés in us. 
-an the thief

**Reply by Ailol** [reply]
not "wingman on snapchat" but someone else named the same for the east hats. a wierd dude tbh(sorry, needs to b said).
